### ISSUE TITLE
Duplicate "fix with assistant" code action

### ISSUE THREAD

**Issue by WeetHet** [zed-industries/zed#22400]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Install Zed Preview (the issue is not present on Stable), open any project with an error, open code actions, observe the issue

### Zed Version and System Specs

Zed: v0.168.0 (Zed Nightly 564936e1fee313efcd4643c5e78472f6dc44ffc5)
OS: macOS 15.2.0
Memory: 64 GiB
Architecture: aarch64

### If applicable, add screenshots or screencasts of the incorrect state / behavior

<img width="834" alt="Screenshot 2024-12-24 at 19 16 37" src="https://github.com/user-attachments/assets/db815531-ccb8-4c6b-bd5b-2c1c5e8cb3f9" />

**Comment by notpeter** [comment]
I can reproduce this on Preview/Nightly.

**Comment by jackTabsCode** [comment]
@notpeter  May have something to do with the fact that there are two "assistant" modules (assistant and assistant2) running at the same time? Not sure, just a guess based on code search results.

**Comment by Tem3dy** [comment]
Was this really fixed?
![image](https://github.com/user-attachments/assets/cbc025df-e423-4769-b4a5-74c7a3d4da84)
I have observed the same issue occuring while coding in C++, couldn't reproduce this specific one, but I've had up to 4 to 5 "Fix with Assistant' code actions, I'm on v0.168.3

**Comment by SomeoneToIgnore** [comment]
It was, just 5 days ago, wile Zed's releases happen on Wednesdays.
0.168.x is Stable, so the fix arrives there in 2 weeks.

**Comment by WeetHet** [comment]
> It was, just 5 days ago, wile Zed's releases happen on Wednesdays. 0.168.x is Stable, so the fix arrives there in 2 weeks.

I'd prefer if it was cherry-picked into stable. Users interact with code actions very frequently and this might get pretty annoying

**Comment by SomeoneToIgnore** [comment]
Fair enough, I've picked it into 0.169.x which will become Stable this Wednesday.
